common/test/acceptance/tests/lms/test_lms_dashboard.py

```diff
@@ -81,3 +86,154 @@ def test_dashboard_social_sharing_feature(self):
+        # otherwise we leave it to the default values
+        self.course_start_date = None
+        self.course_end_date = None
+
+        # now datetime for usage in tests
+        self.now = datetime.datetime.now()
+
+        # Some parameters are provided by the parent setUp() routine, such as the following:
+        # self.course_id, self.course_info, self.unique_id
+        super(LmsDashboardCourseDateTest, self).setUp()
+
+        # Load page objects for use by the tests
+        self.dashboard_page = DashboardPage(self.browser)
+
+    def course_and_user_setup(self):
+        """
+        Create a Course with start and end dates and then create a test user and enroll the user into the course,
+        After the user is logged in and enrolled in the course, visit dashboard page.
+        """
+        # install course fixtures with start and end dates
+        self.install_course_fixture(
+            self.course_info["org"],
+            self.course_info["number"],
+            self.course_info["run"],
+            self.course_info["display_name"],
+            start_date=self.course_start_date,
+            end_date=self.course_end_date,
+        )
+
+        # create a user and enroll them in the course above created
+        self.create_user_and_enroll_them_with_course(
+            username=self.username,
+            email=self.email,
+            course_id=self.course_id
+        )
+
+        # Navigate the authenticated, enrolled user to the dashboard page and get testing!
+        self.dashboard_page.visit()
+
+    def install_course_fixture(self, org, number, run, display_name, start_date=None, end_date=None, settings=None):
+        """
+        Configure some aspects of the test course and install the settings into the course
+        """
+
+        course_fixture = CourseFixture(
+            org=org,
+            number=number,
+            run=run,
+            display_name=display_name,
+            start_date=start_date,
+            end_date=end_date,
+            settings=settings,
+        )
+        course_fixture.install()
+
+    def create_user_and_enroll_them_with_course(self, username, email, course_id):
+        """
+        Create the test user, register them for the course, and authenticate
+        """
+
+        auto_auth_page = AutoAuthPage(
+            self.browser,
+            username=username,
+            email=email,
+            course_id=course_id,
+        )
+        auto_auth_page.visit()
+
+    def test_ended_course_date(self):
+        """
+        Validate the behavior of course dates on student dashboard, when course has ended
+        """
+        self.course_start_date = datetime.datetime(1970, 1, 1)
+        self.course_end_date = self.now - datetime.timedelta(days=90)
+
+        self.course_and_user_setup()
+
+        end_date = self.course_end_date.strftime(DEFAULT_SHORT_DATE_FORMAT)
+        expected_course_date = "Ended - {end_date}".format(end_date=end_date)
```

Comment:
These expression should use same functions as used in student dashboard view i.e `course_overview.start_datetime_text` instead of duplicating them here.